content/en/blog/custom-metrics-autoscaling-new-deploy.md

```diff
@@ -0,0 +1,596 @@
+It will create an environment, a package, a newdeploy function and a kafka mqtrigger.
+
+We need to get the uid of the mqtrigger which is also the name of the `consumergroup`.
+
+Run the command `kubectl get messagequeuetriggers.fission.io -oyaml` and copy the `uid` field value which is under `metadata`.
+
+## Setting up Prometheus adapter
+
+We have kafka and prometheus both up and running, but we need an adapter to expose the custom metrics to the HPA in our newdeploy function.
+So we'll install the [prometheus adapter](https://artifacthub.io/packages/helm/prometheus-community/prometheus-adapter) using helm with the provided configuration file.
+
+We'll be using the `kafka_consumergroup_lag` metric to determine if the HPA should scale or not.
+
+Save the file as `prometheus-adapter.yaml`.
+
+```yaml
+prometheus:
+  port: 9090
+  url: http://prometheus-operated.monitoring.svc.cluster.local
+  
+rules:
+  default: false
+  resource: {}
+  custom:
+  - seriesQuery: 'kafka_consumergroup_lag'
+    resources:
+      overrides:
+        kubernetes_namespace: {resource: "namespace"}
+        kubernetes_pod_name: {resource: "pod"}
+    name:
+      matches: "kafka_consumergroup_lag"
+      as: "kafka_consumergroup_lag"
+    #Change consumergroup value
+    metricsQuery: 'avg_over_time(kafka_consumergroup_lag{topic="request-topic",consumergroup="3f665dc7-6187-4593-8b81-7e4bb08f7f11"}[1m])'
+```
+
+Before installing, you'll need to change the `consumergroup` file with the uid you copied earlier.
+You'll find the filter in the `metricsQuery` field.
+
+```bash
+helm install my-release prometheus-community/prometheus-adapter -f prometheus-adapter.yaml --namespace monitoring
+```
+
+If this installed correctly, you should see the metric and its value.
+
+```bash
+kubectl get --raw /apis/custom.metrics.k8s.io/v1beta1/namespaces/kafka/pods/*/kafka_consumergroup_lag
+{"kind":"MetricValueList","apiVersion":"custom.metrics.k8s.io/v1beta1","metadata":{"selfLink":"/apis/custom.metrics.k8s.io/v1beta1/namespaces/kafka/pods/%2A/kafka_consumergroup_lag"},"items":[{"describedObject":{"kind":"Pod","namespace":"kafka","name":"my-cluster-kafka-exporter-55867498c9-pnqhz","apiVersion":"/v1"},"metricName":"kafka_consumergroup_lag","timestamp":"2022-05-09T12:35:58Z","value":"0","selector":null}]}
+```
+
+**NOTE**: If you are using a shell different from bash(e.g. zsh), then this might not work. Try using the following command in that scenario.
+
+```bash
+kubectl get --raw /apis/custom.metrics.k8s.io/v1beta1/namespaces/kafka/pods/%2A/kafka_consumergroup_lag
+```
+
+**Note**: If you are not getting any value, it maybe because the metric has not been defined yet. So you'll have to send some messages to the queue.
+
+## Testing
+
+Run a producer function to send 10000 messages to the topic `request-topic` and check the namespace `fission-function` where the new deploy pods will be created or destroyed according to the metric value.
```

Comment:
Add a conclusion section and want more section with a link to the example repo. Refer to previous [blog post](https://fission.io/blog/fission-serverless-function-zapier-webhook-automate-your-workflows/).